File: src/main/java/org/pixeltime/enchantmentsenhance/command/ItemCommand.java
package org.pixeltime.enchantmentsenhance.command;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.event.blackspirit.Enhance;
import org.pixeltime.enchantmentsenhance.listener.MenuHandler;
import org.pixeltime.enchantmentsenhance.manager.ItemManager;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class ItemCommand extends SubCommand {
    @Override
    public void onCommand(Player player, String[] args) {
        if (args.length == 2) {
            if (args[0].equalsIgnoreCase("upgrade")) {
                try {
                    Enhance.enhanceSuccess(player.getItemInHand(), player, true, Integer.parseInt(args[1]) - 1);
                } catch (Exception ex) {
                    player.sendMessage(SettingsManager.config.getString("Config.invalidCommand"));
                }
            } else if (args[0].equalsIgnoreCase("setname")) {
                ItemStack item = player.getItemInHand();
                int level = ItemManager.getItemEnchantLevel(item);
                ItemStack curr = ItemManager.setName(item, ChatColor.translateAlternateColorCodes('&', args[1]));
                player.getInventory().removeItem(item);
                ItemManager.forgeItem(player, curr, level);
            }
        }
    }

    @Override
    public String name() {
        return "item";
    }

    @Override
    public String info() {
        return "&6/enhance item { {upgrade} {level} } | {setname} {name}} &7- "
                + SettingsManager.lang.getString(
                "Help.item");
    }

    @Override
    public String[] aliases() {
        return new String[]{"item", "wupin", "wp"};
    }

    @Override
    public String getPermission() {
        return "Enchantmentsenhance.item";
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/command/player/ItemCommand.kt
package org.pixeltime.enchantmentsenhance.command.player

import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.pixeltime.enchantmentsenhance.command.SubCommand
import org.pixeltime.enchantmentsenhance.event.blackspirit.Lore
import org.pixeltime.enchantmentsenhance.manager.ItemManager
import org.pixeltime.enchantmentsenhance.manager.SettingsManager
import org.pixeltime.enchantmentsenhance.util.Util

class ItemCommand : SubCommand() {

    override val permission: String
        get() = "Enchantmentsenhance.item"

    override fun onCommand(player: Player, args: Array<String>) {
        if (args.size >= 2) {
            when {
                args[0].equals("upgrade", ignoreCase = true) -> try {
                    var item = player.itemInHand
                    val level = ItemManager.getItemEnchantLevel(item)
                    val aimingLevel = Integer.parseInt(args[1])
                    if (level < aimingLevel) {
                        for (i in level + 1..aimingLevel) {
                            item = ItemManager.forgeItem(player, item, i, true)
                        }
                    } else if (aimingLevel < level) {
                        Util.sendMessage(SettingsManager.lang.getString("Config.invalidNumber"), player)
                    }
                } catch (ex: Exception) {
                    Util.sendMessage(SettingsManager.lang.getString("Config.invalidCommand"), player)
                }
                args[0].equals("setname", ignoreCase = true) -> {
                    val item = player.itemInHand
                    val level = ItemManager.getItemEnchantLevel(item)
                    val curr = ItemManager.setName(item, ChatColor.translateAlternateColorCodes('&', args[1]))
                    player.inventory.removeItem(item)
                    ItemManager.forgeItem(player, curr, level, true)
                }
                args[0].equals("lore", ignoreCase = true) -> when {
                    args[1].equals("unbound", ignoreCase = true) -> Lore.removeLore(player.itemInHand)
                    args[1].equals("tradeable", ignoreCase = true) -> Lore.addLore(player.itemInHand, SettingsManager.lang.getString("Lore.tradeableLore"), true)
                    args[1].equals("untradeable", ignoreCase = true) -> Lore.addLore(player.itemInHand, SettingsManager.lang.getString("Lore.untradeableLore"), false)
                }
            }
        } else {
            Util.sendMessage(SettingsManager.lang.getString(
                    "Config.invalidNumber"), player)
        }
    }


    override fun name(): String {
        return "item"
    }

    override fun usage(): String {
        return "/enhance item { {upgrade} {level} } | {setname} {name} | {lore} { {tradeable | untradeable | unbound} }"
    }

    override fun aliases(): Array<String> {
        return arrayOf("item", "wupin", "wp")
    }
}
